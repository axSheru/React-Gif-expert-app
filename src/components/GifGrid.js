import React from 'react'

export const GifGrid = ({ category }) => {

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=The+mandalorian&limit=10&api_key=u1775mXvH3OtSHfVaM2EIIfeO4XCxjp9';
        const resp = await fetch( url );
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url,
            };
        });

        console.log(gifs);
    };

    getGifs();

    return (
        <div>
            <h3>{ category }</h3>
        </div>
    )
}
