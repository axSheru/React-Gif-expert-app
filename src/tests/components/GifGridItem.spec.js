import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en componente GifGridItem.', () => {
    
    let wrapper;
    const title = 'Soy un tÃ­tulo';
    const url = 'https://localhost:8080';

    beforeEach(() => {
        wrapper = shallow( <GifGridItem title={ title } url={ url } /> );
    });

    test('debe de hacer match con el snapshot.', () => {

        expect( wrapper ).toMatchSnapshot();

    });

});