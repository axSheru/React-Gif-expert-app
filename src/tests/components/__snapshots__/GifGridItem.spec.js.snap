// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem. debe de hacer match con el snapshot. 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Soy un título"
    src="https://localhost:8080"
  />
  <p>
    Soy un título
  </p>
</div>
`;
